mypy/checkexpr.py

```diff
@@ -896,9 +900,10 @@ def apply_inferred_arguments(self, callee_type: CallableType,
         return self.apply_generic_arguments(callee_type, inferred_args, context)
 
     def check_argument_count(self, callee: CallableType, actual_types: List[Type],
-                             actual_kinds: List[int], actual_names: List[str],
+                             actual_kinds: List[int],
+                             actual_names: Optional[Sequence[Optional[str]]],
                              formal_to_actual: List[List[int]],
-                             context: Context,
+                             context: Optional[Context],
                              messages: Optional[MessageBuilder]) -> bool:
         """Check that there is a value for all required arguments to a function.
```

Comment:
Maybe note in the docstring that if `messages` is given, `context` must also be given?